pkg/storage/bloom/v1/block.go

```diff
@@ -117,11 +117,11 @@ type BlockQuerier struct {
 // will be returned to the pool for efficiency. This can only safely be used
 // when the underlying bloom bytes don't escape the decoder, i.e.
 // when loading blooms for querying (bloom-gw) but not for writing (bloom-compactor).
-func NewBlockQuerier(b *Block, noCapture bool, maxPageSize int) *BlockQuerier {
+func NewBlockQuerier(b *Block, alloc Allocator, maxPageSize int) *BlockQuerier {
 	return &BlockQuerier{
 		block:  b,
 		series: NewLazySeriesIter(b),
-		blooms: NewLazyBloomIter(b, noCapture, maxPageSize),
+		blooms: NewLazyBloomIter(b, alloc, maxPageSize),
```

Comment:
Instead of having to pass the allocator down the chain, what about having the allocator as a pkg-level variable similarly to here?

https://github.com/grafana/loki/pull/13039/files#diff-8572b07b377f1bfae2ec4ffd992cb8d06972b089d2c195996faae0763621a318R18

Reply:
I prefer explicit passing instead of package-global variables. IMO global state should be avoided.